src/commands/rockpaper.rs

```diff
@@ -1,61 +1,46 @@
-// Stolen from https://github.com/MercuryMer/fartcentral/blob/rust/src/commands/rock.rs to test
-
 use rand::Rng;
 use serenity::builder::{CreateCommand, CreateCommandOption};
-use serenity::model::application::{CommandOptionType, ResolvedOption, ResolvedValue};
-
-// now before you look, i think i fucked some things up
-// gimme a break rust hates me and its scary
-
-pub fn run(options: &[ResolvedOption]) -> String {
-    if let Some(ResolvedOption {
-        value: ResolvedValue::String(choice),
-        ..
-    }) = options.first()
-    {
-        let mut rng = rand::thread_rng();
-        let computer_choice = match rng.gen_range(0..2) {
-            0 => "rock",
-            1 => "paper",
-            2 => "scissors",
-            _ => "error",
-        };
-        match *choice {
-            "rock" => match computer_choice {
-                "rock" => "It's a tie! [rock]",
-                "paper" => "You lose! [paper]",
-                "scissors" => "You win! [scissors!]",
-                _ => "error",
-            },
-            "paper" => match computer_choice {
-                "rock" => "You win! [rock!]",
-                "paper" => "It's a tie! [paper!]",
-                "scissors" => "You lose! [scissors]",
-                _ => "error",
-            },
-            "scissors" => match computer_choice {
-                "rock" => "You lose! [rock]",
-                "paper" => "You win! [paper]",
-                "scissors" => "It's a tie! [scissors]",
-                _ => "error",
-            },
-            _ => "Please provide a valid choice",
-        }
-        .to_string()
-    } else {
-        "Please provide a valid choice".to_string()
-    }
-}
+use serenity::model::application::CommandOptionType;
 
 pub fn register() -> CreateCommand {
     CreateCommand::new("rockpaper")
         .description("Play rock paper scissors with purplewood!")
         .add_option(
             CreateCommandOption::new(
-                CommandOptionType::String,
+                CommandOptionType::Integer,
                 "choice",
                 "choices: rock, paper, scissors",
             )
             .required(true),
         )
 }
+
+pub fn run(player: i32, cpu: Option<i32>) -> String {
+    let choices = ["rock", "paper", "scissors"];
+    let player_choice = choices[player as usize % 3];
```

Comment:
Small bug around here, the bot will always make you select rock even if you change the int to a different value including invalid values like 42.

Reply:
You should also probably +1 the value so people dont run into out of bound index issues when they type 3 (most ppl count from 1-3. not 0-2)